Google.PowerShell/Compute/GceDisk.cs

```diff
@@ -312,34 +322,60 @@ public class ResizeGceDiskCmdlet : GceCmdlet
         /// Name of the disk.
         /// </para>
         /// </summary>
-        [Parameter(Position = 2, Mandatory = true), ValidatePattern("[a-z]([-a-z0-9]*[a-z0-9])?")]
+        [Parameter(ParameterSetName = ParameterSetNames.ByName, Position = 0, Mandatory = true)]
+        [ValidatePattern("[a-z]([-a-z0-9]*[a-z0-9])?")]
         public string DiskName { get; set; }
 
         /// <summary>
         /// <para type="description">
+        /// The Disk object that describes the disk to resize.
+        /// </para>
+        /// </summary>
+        [Parameter(ParameterSetName = ParameterSetNames.ByObject, Mandatory = true, ValueFromPipeline = true)]
+        public Disk InputObject { get; set; }
+
+        /// <summary>
+        /// <para type="description">
         /// Specify the new size of the disk in GiB. Must be larger than the current disk size.
         /// </para>
         /// </summary>
-        [Parameter(Position = 3, Mandatory = true)]
+        [Parameter(ParameterSetName = ParameterSetNames.ByName, Mandatory = true, Position = 1)]
+        [Parameter(ParameterSetName = ParameterSetNames.ByObject, Mandatory = true, Position = 0)]
         public long NewSizeGb { get; set; }
 
         protected override void ProcessRecord()
         {
-            base.ProcessRecord();
-
-            DisksResizeRequest diskResizeReq = new DisksResizeRequest();
-            diskResizeReq.SizeGb = NewSizeGb;
+            string project;
+            string zone;
+            string name;
+            switch (ParameterSetName)
+            {
+                case ParameterSetNames.ByName:
+                    project = Project;
+                    zone = Zone;
+                    name = DiskName;
+                    break;
+                case ParameterSetNames.ByObject:
+                    project = GetProjectNameFromUri(InputObject.SelfLink);
+                    zone = GetZoneNameFromUri(InputObject.Zone);
+                    name = InputObject.Name;
+                    break;
+                default:
+                    throw UnknownParameterSetException;
+            }
 
-            DisksResource.ResizeRequest resizeReq = Service.Disks.Resize(diskResizeReq, Project, Zone, DiskName);
+            var diskResizeReq = new DisksResizeRequest { SizeGb = NewSizeGb };
+            DisksResource.ResizeRequest resizeReq =
+                Service.Disks.Resize(diskResizeReq, project, zone, name);
 
             Operation op = resizeReq.Execute();
-            WaitForZoneOperation(Project, Zone, op);
-
-            // Return the updated disk.
-            DisksResource.GetRequest getReq = Service.Disks.Get(Project, Zone, DiskName);
-            Disk disk = getReq.Execute();
-
-            WriteObject(disk);
+            AddZoneOperation(project, zone, op, () =>
+            {
+                // Return the updated disk.
```

Comment:
This comment isn't really needed. It says what the code does (which is pretty obvious), it doesn't help explain why the code does it (which isn't necessary).